### ISSUE TITLE
Does async work in IE11? (slow performance with React)

### ISSUE THREAD

**Issue by gitgrimbo** [PrismJS/prism#1441]
I have a code block that is taking ~10 seconds to highlight, and when I use the following code (using async set to `true`) I can't scroll the window while the highlighting takes place.

```js
    let now = performance.now();
    Prism.highlightElement(code, true, () => console.log(performance.now() - now));
```

**Comment by gitgrimbo** [comment]
I think the reason IE11 takes so long might have something to do with React (currently using v16.4.0).

The "set innerHTML" row for the first load is 57**ms**, and the reload is 10.45**s**.

Performance trace from IE11 on first load of the page with highlighting:

![image](https://user-images.githubusercontent.com/1020541/41337012-e63440ca-6ee5-11e8-9e2b-b43b1c044236.png)

Performance trace from IE11 on *a reload* of the page with highlighting:

![image](https://user-images.githubusercontent.com/1020541/41337036-f4f57516-6ee5-11e8-80ba-ff4f781cb518.png)

The JavaScript call stacks of a different, but also slow, render. It looks like `_self.Prism.highlight` (and subtree) doesn't take much time, and the culprit lies somewhere else within `_self.Prism.highlightElement` - probably the assignment to `innerHTML` if we use the previous chart as evidence (https://github.com/PrismJS/prism/blob/435eb2e6cd94948627b044255aab7605235bdfa5/components/prism-core.js#L261..L270).

![image](https://user-images.githubusercontent.com/1020541/41338096-0b0f3758-6ee9-11e8-8855-63137fd250cf.png)

**Comment by gitgrimbo** [comment]
I created a fiddle as an example

https://jsfiddle.net/fiddlegrimbo/vhgj510p/20/show

Consistently highlights the HTML in ~500ms on IE11.  So there is something going on in my webpack/React code to slow the highlighting down.

**Comment by mAAdhaTTah** [comment]
Definitely related to `innerHTML`, based on your charts, but if you get it @ 500ms consistently normally, but 10s when used w/ React / Webpack, that would suggest it's something about the interaction between those two that's causing the issue. I'll leave this open for other to weigh in, but I'm not sure we can do anything on our end.

**Comment by gitgrimbo** [comment]
Thanks for your reply.

I'm starting to suspect IE11 devtools now. I restarted IE11 and did not open devtools, and both my vanilla test of Prism and my React app are highlighting quickly.

When I open devtools it slows to a crawl.

This slowdown can be reproduced using the test page, https://prismjs.com/test.html, and the content of this gist, https://gist.github.com/gitgrimbo/3ae03923ae1fb3b4865895c07a6e65f2#file-html-html.

When IE11 devtools is not open, this highlights for me in less than a second. With devtools open, it highlights in ~9 seconds.

So in summary I don't think this is a real issue with Prism, although it might be interesting to know why IE11 devtools could cause such a slowdown.

**Comment by mAAdhaTTah** [comment]
Performance profiling itself _can_ cause slowness, as it has to instrument the code in a way that can impact how fast it runs, but that's the first time I've seen the impact be that large. Admittedly, I'm not entirely surprised IE11's devtools aren't great.

**Comment by mAAdhaTTah** [comment]
Alright, I think we're ok to close this then, but if you find an issue here we should look it, please reopen. Thanks!
